src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/InputEmptyCase.java

```diff
@@ -0,0 +1,36 @@
+////////////////////////////////////////////////////////////////////////////////
+// Input test file for testing empty LITERAL_CASE.
+// Created: 2017
+////////////////////////////////////////////////////////////////////////////////
+package com.puppycrawl.tools.checkstyle.checks.blocks;
+
+class InputEmptyCase
+{
+    void method1(int a) {
+        switch (a) {}
+        switch (a) {case 1: ; } // no violation
+        switch (a) {case 1:{}}  // 1 violation
+        switch (a) {
+            case 1:
+        }
+        switch (a) {
+            case 1: // 1 violation
+            {}
+        }
+        switch (a) {
+            case 1: // 1 violation if checking statements
+            {// none if checking text
+            }
+        }
+    }
+
+    public void method2(char c) {
+        switch(c) { case 0: } // no violation
+        switch(c) { case 0: {} } // 1 violation
+        switch(c) { case 0: method1(0); {} } // no violation
+        switch(c) { case 0: case 1: {} } // 1 violation
+        switch(c) { case 0: {} case 1: {// 2 violations if checking statements
+        }
+        }
+    }
+}
```

Comment:
@romani yes the blocks are all identical and if we were just verifying blocks then no special processing would have been necessary. We are investigating empty case. Hence all the blocks inside the case need to be checked. for example:
   
    case 1: {  } {//something         //NO violation
    } {System.out.println(); { } }         //NO violation

Now I need to loop through all the blocks in the case to determine if case is empty. (I can't say case is empty looking at just one of the blocks). And that's all I am doing in `isCaseEmpty`